src/app/calculator/furnaces/flue-gas/flue-gas-form-volume/flue-gas-form-volume.component.ts

```diff
@@ -0,0 +1,286 @@
+      updatedFlueGasData = currentBaseline;
+    } else {
+      let currentModification: FlueGas = this.flueGasService.modificationData.getValue();
+      updatedFlueGasData = currentModification;
+    }
+
+    if (updatedFlueGasData.flueGasByVolume) {
+      this.byVolumeForm = this.flueGasService.initByVolumeFormFromLoss(updatedFlueGasData);
+    } else {
+      this.byVolumeForm = this.flueGasService.initEmptyVolumeForm();
+    }
+    this.initFormSetup();
+  }
+
+  checkWarnings() {
+    let tmpLoss: FlueGasByVolume = this.flueGasService.buildByVolumeLossFromForm(this.byVolumeForm);
+    this.warnings = this.flueGasService.setByVolumeWarnings(tmpLoss);
+  }
+
+  setCalcMethod() {
+    if (this.byVolumeForm.controls.oxygenCalculationMethod.value === 'Excess Air') {
+      this.calcMethodExcessAir = true;
+    } else {
+      this.calcMethodExcessAir = false;
+    }
+  }
+  
+  calcExcessAir() {
+    let input = {
+      CH4: this.byVolumeForm.controls.CH4.value,
+      C2H6: this.byVolumeForm.controls.C2H6.value,
+      N2: this.byVolumeForm.controls.N2.value,
+      H2: this.byVolumeForm.controls.H2.value,
+      C3H8: this.byVolumeForm.controls.C3H8.value,
+      C4H10_CnH2n: this.byVolumeForm.controls.C4H10_CnH2n.value,
+      H2O: this.byVolumeForm.controls.H2O.value,
+      CO: this.byVolumeForm.controls.CO.value,
+      CO2: this.byVolumeForm.controls.CO2.value,
+      SO2: this.byVolumeForm.controls.SO2.value,
+      O2: this.byVolumeForm.controls.O2.value,
+      o2InFlueGas: this.byVolumeForm.controls.o2InFlueGas.value,
+      excessAir: this.byVolumeForm.controls.excessAirPercentage.value
+    };
+
+    if (!this.calcMethodExcessAir) {
+      if (this.byVolumeForm.controls.o2InFlueGas.status === 'VALID') {
+        this.calculationExcessAir = this.phastService.flueGasCalculateExcessAir(input);
+        this.byVolumeForm.patchValue({
+          excessAirPercentage: this.calculationExcessAir,
+        });
+      } else {
+        this.calculationExcessAir = 0;
+        this.byVolumeForm.patchValue({
+          excessAirPercentage: this.calculationExcessAir,
+        });
+      }
+    }
+
+    if (this.calcMethodExcessAir) {
+      if (this.byVolumeForm.controls.excessAirPercentage.status === 'VALID') {
+        this.calculationFlueGasO2 = this.phastService.flueGasCalculateO2(input);
+        this.byVolumeForm.patchValue({
+          o2InFlueGas: this.calculationFlueGasO2,
+        });
+      } else {
+        this.calculationFlueGasO2 = 0;
+        this.byVolumeForm.patchValue({
+          o2InFlueGas: this.calculationFlueGasO2,
+        });
+      }
+    }
+    this.calculate();
+  }
+  
+  calculate() {
+    this.checkFlueGasTemp();
+    this.checkWarnings();
+    if (this.isBaseline) {
+      let currentBaselineData: FlueGas = this.flueGasService.baselineData.getValue();
+      let currentBaselineByVolume: FlueGasByVolume = this.flueGasService.buildByVolumeLossFromForm(this.byVolumeForm)
```

Comment:
Unneeded extra variable here too. Same in other form, just going to comment here.